### ISSUE TITLE
Error = Mistake ?

### ISSUE THREAD

**Issue by acse-c2434dbb** [ese-msc/introduction-to-python#19]
To whom it may concern, 

I hope this finds you well. 
I am wondering does pybryt error necessarily imply mistake in the code?
In exercise 1.13, I got a pybryt error '_ERROR: Hmmm... you are not appending an odd number to the list in each iteration_'. However, my function does append odd numbers in the list and successfully passes okpy test. 
Please could you help advise how I could solve the error. 

Thank you very much !

**Comment by marijanbeg** [comment]
Hi @acse-c2434dbb, thank you for your question. Yes, it is possible that PyBryt complains about your solution even if it is correct. These types of mistakes are very valuable to us because they allow us to improve our tests and reference solutions. Could you please do the following for us:

1. Restart you JupyterHub session (this will ensure you are using the newest version of PyBryt)
2. Run the exercise code again
3. If the error message is still there, please send us your solution code and we will investigate it.

Thank you very much for your feedback, and we are looking forward to hearing from you.

**Comment by acse-c2434dbb** [comment]
Good afternoon, @marijanbeg. 
Thank you for your quick reply. 
I have tried to restart the session and rerun the code. The error messages still appear. Could you please give me your contact so that I could send my codes over to you? 
Thank you very much!

**Comment by marijanbeg** [comment]
Of course, you can find my email on this webpage: https://www.imperial.ac.uk/people/m.beg

**Comment by acse-c2434dbb** [comment]
@marijanbeg I have just sent you an email. Thank you very much !

**Comment by marijanbeg** [comment]
Hi @acse-c2434dbb, thank you for sending me your solutions. I think you are overusing `while` loops - they are more difficult to write and it is much easier to accidentally introduce an error.

Could you please attempt to solve these exercises using for-loops? Please note that you can use `range(start, stop, step)` in loops like `for i in range(start, stop, step)` to iterate over the elements of interest.

Please feel free to send me your solutions again and I will have another look.

**Comment by acse-c2434dbb** [comment]
@marijanbeg Thank you very much for your suggestions. I will try to solve them again.

**Comment by edsml-f80c6dfd** [comment]
Hi, @marijanbeg

I got the same problem _'ERROR: Hmmm... you are not appending an odd number to the list in each iteration'_ in **Exercise 1.13**. Here is my code. Could you please help me figure out the problem? Thanks!!

- using while loop

```
def odd_numbers(n):
    L = []
    i = 1
    while i <= n:
        L.append(i)
        i = i + 2
    return L
```

- using for-loop

```
def odd_numbers(n):
    L = []
    i = 1
    for i in range(i,n+1,2):
        L.append(i)
    return L
```
